lib/puppet/util/run_mode.rb

```diff
@@ -85,21 +85,21 @@ def code_dir
       end
 
       def var_dir
-        which_dir(File.join(windows_common_base("var")), "~/.puppet/var")
+        which_dir(File.join(windows_common_base("puppet/cache")), "~/.puppet/var")
```

Comment:
Puppet will create any directory that is a puppet directory-based setting, which includes `vardir` (cache), `confdir`, `logdir`, and `rundir`.

However, the MSI must create directories that are not puppet settings, `C:\ProgramData\PuppetLabs\puppet\var`, including parent directories of things that are settings, e.g. `C:\ProgramData\PuppetLabs\puppet`